Index: sagemaker-code-editor/vscode/src/vs/server/node/webClientServer.ts
===================================================================
--- sagemaker-code-editor.orig/vscode/src/vs/server/node/webClientServer.ts
+++ sagemaker-code-editor/vscode/src/vs/server/node/webClientServer.ts
@@ -102,6 +102,7 @@ export class WebClientServer {
 	private readonly _callbackRoute: string;
 	private readonly _webExtensionRoute: string;
 	private readonly _idleRoute: string;
+	private readonly _hello: string;
 
 	constructor(
 		private readonly _connectionToken: ServerConnectionToken,
@@ -118,6 +119,7 @@ export class WebClientServer {
 		this._callbackRoute = `${serverRootPath}/callback`;
 		this._webExtensionRoute = `${serverRootPath}/web-extension-resource`;
 		this._idleRoute = '/api/idle';
+		this._hello= '/api/hello';
 	}
 
 	/**
@@ -138,6 +140,9 @@ export class WebClientServer {
 			if (pathname === this._idleRoute) {
 				return this._handleIdle(req, res);
 			}
+			if (pathname === this._hello) {
+				return this._handleHello(req, res);
+			}
 			if (pathname === this._callbackRoute) {
 				// callback support
 				return this._handleCallback(res);
@@ -482,6 +487,13 @@ export class WebClientServer {
 			serveError(req, res, 500, error.message)
 		}
 	}
+
+	private async _handleHello(req: http.IncomingMessage, res: http.ServerResponse): Promise<void> {
+		res.statusCode = 200;
+		res.setHeader('Content-Type', 'application/json');
+		res.end(JSON.stringify({ 'msg': 'hello' }));
+	}
 }
 
 /**
